pkg/insights/insightsclient/insightsclient.go

```diff
@@ -113,6 +122,39 @@ func clientTransport() http.RoundTripper {
 	return transport.DebugWrappers(clientTransport)
 }
 
+// ConfigProxier is creating a Proxier from proxy set in HttpConfig
+func ConfigProxier(c config.HTTPConfig) func(req *http.Request) (*url.URL, error) {
+	proxyConfig := httpproxy.Config{
+		HTTPProxy:  c.HTTPProxy,
+		HTTPSProxy: c.HTTPSProxy,
+		NoProxy:    c.NoProxy,
+	}
+	// The golang ProxyFunc seems to have NoProxy already built in
+	return func(req *http.Request) (*url.URL, error) {
+		return proxyConfig.ProxyFunc()(req.URL)
+	}
+}
+
+// FromConfig is setting HttpProxy from HttpConfig in support secret, if it is used
+func FromConfig(c config.HTTPConfig) func(req *http.Request) (*url.URL, error) {
```

Comment:
`ProxyFromConfig` as name

Reply:
Actually, these two methods are confusingly named in general.  This should probably be `NewProxyFromConfig()` in the line above.

Reply:
I removed them when I moved the logic to authorizer and removed options to simplify a bit.